client/mysqldump.c

```diff
@@ -2127,6 +2138,25 @@ static char *quote_name(const char *name, char *buff, my_bool force)
   return buff;
 } /* quote_name */
 
+static char *quote_name_for_header(const char *name, char *buff, my_bool force)
```

Comment:
I don't see any tests that really test this function. Should fields-enclosed-by, fields-separated-by and lines-terminated-by characters be escaped?

Reply:
I am sorry, but I did not understand this point. Can you please give me some more detail?

Reply:
This is quoting header names, but all your tests have very basic ascii column names that don't need quoting. What does it look like if these have quotes or escape characters in them.

Reply:
The new version of the test includes a new column on the table with quotes and diacritics in its name.